src/sentry/migrations/0561_backfill_new_notification_tables.py

```diff
@@ -213,77 +213,81 @@ class NotificationScopeType(Enum):
 
 
 def backfill_notification_settings(apps, schema_editor):
-    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
-    NotificationSettingOption = apps.get_model("sentry", "NotificationSettingOption")
-    NotificationSettingProvider = apps.get_model("sentry", "NotificationSettingProvider")
-
-    for setting in RangeQuerySetWrapperWithProgressBar(NotificationSetting.objects.all()):
-        # find all the related settings regardless of the provider
-        # note that we will end up setting the settings N times for N provider options
-        related_settings = NotificationSetting.objects.filter(
-            scope_type=setting.scope_type,
-            scope_identifier=setting.scope_identifier,
-            user_id=setting.user_id,
-            team_id=setting.team_id,
-            type=setting.type,
-        )
-
-        enabled_providers = []
-        all_providers = []
-        enabled_value = None
-        for related_setting in related_settings:
-            if related_setting.value != NotificationSettingOptionValues.NEVER.value:
-                enabled_providers.append(related_setting.provider)
-                enabled_value = related_setting.value
-            all_providers.append(related_setting.provider)
-
-        update_args = {
-            "type": get_notification_setting_type_name(related_setting.type),
-            "user_id": related_setting.user_id,
-            "team_id": related_setting.team_id,
-            "scope_type": get_notification_scope_name(related_setting.scope_type),
-            "scope_identifier": related_setting.scope_identifier,
-        }
-
-        # check if all are disabled
-        if len(enabled_providers) == 0:
-            NotificationSettingOption.objects.update_or_create(
-                **update_args, defaults={"value": NotificationSettingsOptionEnum.NEVER.value}
-            )
-        else:
-            # map the enabled setting
-            NotificationSettingOption.objects.update_or_create(
-                **update_args,
-                defaults={
-                    "value": NOTIFICATION_SETTING_OPTION_VALUES[
-                        NotificationSettingOptionValues(enabled_value)
-                    ]
-                },
+    try:
```

Comment:
had to modify this migration because for some reason it errors unexpectedly in the test_two_archived_with_same_name test